tests/test_co2brine_ptflash.cpp

```diff
@@ -0,0 +1,146 @@
+    //fs.setMoleFraction(FluidSystem::oilPhaseIdx, FluidSystem::Comp2Idx, comp[2]);
+
+    fs.setMoleFraction(FluidSystem::gasPhaseIdx, FluidSystem::Comp0Idx, comp[0]);
+    fs.setMoleFraction(FluidSystem::gasPhaseIdx, FluidSystem::Comp1Idx, comp[1]);
+    //fs.setMoleFraction(FluidSystem::gasPhaseIdx, FluidSystem::Comp2Idx, comp[2]);
+
+    // It is used here only for calculate the z
+    fs.setSaturation(FluidSystem::oilPhaseIdx, sat[0]);
+    fs.setSaturation(FluidSystem::gasPhaseIdx, sat[1]);
+
+    fs.setTemperature(temp);
+
+    // ParameterCache paramCache;
+    {
+        typename FluidSystem::template ParameterCache<Evaluation> paramCache;
+        paramCache.updatePhase(fs, FluidSystem::oilPhaseIdx);
+        paramCache.updatePhase(fs, FluidSystem::gasPhaseIdx);
+        fs.setDensity(FluidSystem::oilPhaseIdx, FluidSystem::density(fs, paramCache, FluidSystem::oilPhaseIdx));
+        fs.setDensity(FluidSystem::gasPhaseIdx, FluidSystem::density(fs, paramCache, FluidSystem::gasPhaseIdx));
+    }
+
+    ComponentVector zInit(0.); // TODO; zInit needs to be normalized.
+    {
+        Scalar sumMoles = 0.0;
+        for (unsigned phaseIdx = 0; phaseIdx < FluidSystem::numPhases; ++phaseIdx) {
+            for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
+                Scalar tmp = Opm::getValue(fs.molarity(phaseIdx, compIdx) * fs.saturation(phaseIdx));
+                zInit[compIdx] += Opm::max(tmp, 1e-8);
+                sumMoles += tmp;
+            }
+        }
+        zInit /= sumMoles;
+        // initialize the derivatives
+        // TODO: the derivative eventually should be from the reservoir flow equations
+        Evaluation z_last = 1.;
+        for (unsigned compIdx = 0; compIdx < numComponents - 1; ++compIdx) {
+            zInit[compIdx] = Evaluation::createVariable(Opm::getValue(zInit[compIdx]), compIdx + 1);
+            z_last -= zInit[compIdx];
+        }
+        zInit[numComponents - 1] = z_last;
+    }
+
+    // TODO: only, p, z need the derivatives.
+    const double flash_tolerance = 1.e-12; // just to test the setup in co2-compositional
+    const int flash_verbosity = 1;
+    //const std::string flash_twophase_method = "ssi"; // "ssi"
+    //const std::string flash_twophase_method = "newton";
+     const std::string flash_twophase_method = "newton";
+
+    // TODO: should we set these?
+    // Set initial K and L
+    for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
+        const Evaluation Ktmp = fs.wilsonK_(compIdx);
+        fs.setKvalue(compIdx, Ktmp);
+    }
+    const Evaluation Ltmp = 1.;
+    fs.setLvalue(Ltmp);
+
+    const int spatialIdx = 0;
+    using Flash = Opm::PTFlash<double, FluidSystem>;
+    // TODO: here the zInit does not have the proper derivatives
+    Flash::solve(fs, zInit, spatialIdx, flash_verbosity, flash_twophase_method, flash_tolerance);
+
+}
+
+int main(int argc, char **argv)
+{
+    Dune::MPIHelper::instance(argc, argv);
+
+    testCo2BrineFlash();
```

Comment:
This test appears to be restricted to "compile + run" only.  How hard would it be to add checks for "produces expected result" too?

Reply:
Yeah, we should be able to check the results.